File: msNote/src/main/java/com/Jirawah/msNote/service/NoteService.java
package com.Jirawah.msNote.service;

import com.Jirawah.msNote.dto.NoteRequestDTO;
import com.Jirawah.msNote.dto.NoteResponseDTO;

import java.util.List;

public interface NoteService {

    NoteResponseDTO createNote(NoteRequestDTO requestDTO);

    List<NoteResponseDTO> getNotesByPatientId(String patientId);

    NoteResponseDTO updateNote(String noteId, NoteRequestDTO requestDTO);

    void deleteNote(String noteId);
}

File: webApp-frontend/src/app/service/note.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Note } from '../model/note.model';

@Injectable({
  providedIn: 'root'
})
export class NoteService {

  private apiUrl = 'http://localhost:8085/notes';

  constructor(private http: HttpClient) {}

  private getAuthHeaders(): HttpHeaders {
    const token = localStorage.getItem('token');
    return new HttpHeaders({
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${token}`
    });
  }

  getNotesByPatientId(patientId: string): Observable<Note[]> {
    return this.http.get<Note[]>(`${this.apiUrl}/patient/${patientId}`, {
      headers: this.getAuthHeaders()
    });
  }

  createNote(note: Partial<Note>): Observable<Note> {
    return this.http.post<Note>(this.apiUrl, note, {
      headers: this.getAuthHeaders()
    });
  }

  updateNote(noteId: string, note: Partial<Note>): Observable<Note> {
    return this.http.put<Note>(`${this.apiUrl}/${noteId}`, note, {
      headers: this.getAuthHeaders()
    });
  }

  deleteNote(noteId: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${noteId}`, {
      headers: this.getAuthHeaders()
    });
  }
}